Content.Server/Paper/PaperSystem.cs

```diff
@@ -100,6 +111,44 @@ private void OnExamined(EntityUid uid, PaperComponent paperComp, ExaminedEvent a
             }
         }
 
+        private void UpdateTagsState(EntityUid userUid, PaperComponent paperComp)
+        {
+            if (paperComp.TagsState != null)
+            {
+                paperComp.TagsState = TryComp<MetaDataComponent>(userUid, out var metaEntity)
+                    ? paperComp.TagsState with { PersonName = metaEntity.EntityName }
+                    : paperComp.TagsState with { PersonName = Loc.GetString("paper-tags-person-name-default") };
+
+                var station = _station.GetOwningStation(userUid);
+                paperComp.TagsState = station is null
+                    ? paperComp.TagsState with { StationName = Loc.GetString("paper-tags-station-name-default") }
+                    : paperComp.TagsState with { StationName = Name(station.Value) };
+
+                // Searches among the PDA slot, through which the IdCard Com content
+                // containing the position is obtained. This is necessary in order for
+                // the position in the tag to change, unlike the previous method.
+                var defaultJobTitle = Loc.GetString("paper-tags-person-job-default");
+                if (TryComp<InventoryComponent>(userUid, out var inventoryComponent))
+                {
+                    var slots = new InventorySlotEnumerator(inventoryComponent);
+                    while (slots.MoveNext(out var slot))
+                    {
+                        if (slot is { ID: "id", ContainedEntity: not null }
```

Comment:
theres a helper somewhere to find the id card either equipped or in a pda use that instead of this meme

then
```cs
state.PersonJob = Loc.GetString("paper-tags-person-job-default");
if (TheHelper(out var idCard))
    state.PersonJob = idCard.JobTitle;
```